quickfixj-core/src/test/java/quickfix/SocketInitiatorTest.java

```diff
@@ -237,18 +241,21 @@ public void testInitiatorConnectionException() throws Exception {
         // use a free port to make sure nothing is listening
         int freePort = AvailablePortFinder.getNextAvailable();
         Initiator initiator = null;
+        String host = "localhost";
         AtomicBoolean onConnectExceptionWasCalled = new AtomicBoolean(false);
+        AtomicReference<SocketAddress> socketAddress = new AtomicReference<>(null);
         try {
             SessionID clientSessionID = new SessionID(FixVersions.BEGINSTRING_FIX42, "TW", "ISLD");
             SessionSettings settings = getClientSessionSettings(clientSessionID, freePort);
             settings.setString(clientSessionID, "ReconnectInterval", "1");
-            settings.setString(clientSessionID, "SocketConnectHost", "localhost");
-            settings.setString(clientSessionID, "SocketConnectProtocol", ProtocolFactory.getTypeString(ProtocolFactory.VM_PIPE));
+            settings.setString(clientSessionID, "SocketConnectHost", host);
+            settings.setString(clientSessionID, "SocketConnectProtocol", ProtocolFactory.getTypeString(ProtocolFactory.SOCKET));
```

Comment:
Does this not work with `VM_PIPE` addresses or what was the reason for the change?

Reply:
It does work with `VM_PIPE`, but I'm not sure if this is a sufficient test, because I didn't find a lot about this protocol.
If I understand it correctly, it doesn't work over network but uses some other means to communicate between local peers located in the same environment or on the same server

If needed, I can switch it back to `VM_PIPE` (and reduce the sleep delay as well), the assertion in this case will spread only to the port value